src/types/redis_bloom_chain.cc

```diff
@@ -142,79 +152,84 @@ rocksdb::Status BloomChain::Add(const Slice &user_key, const Slice &item, int *r
   std::vector<std::string> bf_key_list;
   getBFKeyList(ns_key, metadata, &bf_key_list);
 
+  std::vector<std::string> bf_data_list;
+  s = getBFDataList(bf_key_list, &bf_data_list);
+  if (!s.ok()) return s;
+
   auto batch = storage_->GetWriteBatchBase();
   WriteBatchLogData log_data(kRedisBloomFilter, {"insert"});
   batch->PutLogData(log_data.Encode());
 
-  // check
-  std::vector<bool> exist{false};                      // TODO: to refine in BF.MADD
-  for (int i = metadata.n_filters - 1; i >= 0; --i) {  // TODO: to test which direction for searching is better
-    s = bloomCheck(bf_key_list[i], {item}, &exist);
-    if (!s.ok()) return s;
-    if (exist[0]) {
-      *ret = 0;
-      break;
+  for (size_t i = 0; i < items.size(); ++i) {
```

Comment:
This part is right, but can we leave a todo here, and change the procedure to:
1. compute hashes for all insert keys ( we don't need these keys actually, hashes is computed multiple times)
2. for each `bf_data`, batch check the input hash if it doesn't exists

I guess this could be 2 times faster than current impl, and avoid multiple `bloomCheck`

Reply:
Emmm I think you're right, but maybe this kind of conflict doesn't matter. 

Once this happened, it would report full or pre-reserve some bloom "slot", but the probability is low, and it merely meets hash collision, so I think it's ok to do this

Reply:
The worst cast might be madd same key multiple times. Maybe we need to handle this case by dedup the hash or other. Here I just want to avoid get a BloomFilter key from storage for multiple times.

Reply:
But now, we just get bloomfilter data at once and put them into a data_list. It seems not to get from storage for multiple times.